source/extensions/transport_sockets/tls/context_impl.cc

```diff
@@ -263,7 +272,7 @@ ContextImpl::ContextImpl(Stats::Scope& scope, const Envoy::Ssl::ContextConfig& c
   }
 
   for (auto& ctx : tls_contexts_) {
-    if (verify_mode != SSL_VERIFY_NONE) {
+    if (verify_mode != SSL_VERIFY_NONE && !capabilities_.verifies_peer_certificates) {
```

Comment:
This can be moved outside of the `for` loop.

Reply:
Done.